stub-runner/stub-runner-spring/src/main/groovy/com/ofg/stub/StubDownloader.groovy

```diff
@@ -0,0 +1,157 @@
+package com.ofg.stub
+
+import com.ofg.stub.spring.ZipCategory
+import groovy.util.logging.Slf4j
+
+import static groovy.grape.Grape.addResolver
+import static groovy.grape.Grape.resolve
+import static groovy.io.FileType.FILES
+import static java.nio.file.Files.createTempDirectory
+
+@Slf4j
+class StubDownloader {
+
+    private static final String LATEST_MODULE = '*'
+    private static final String REPOSITORY_NAME = 'dependency-repository'
+    private static final String STUB_RUNNER_TEMP_DIR_PREFIX = 'stub-runner'
+
+    private DependencyResolver dependencyResolver
+
+    StubDownloader() {
+    }
+
+    StubDownloader(DependencyResolver dependencyResolver) {
+        this.dependencyResolver = dependencyResolver
+    }
+
+    File downloadAndUnpackStubJar(boolean skipLocalRepo, String stubRepositoryRoot, String stubsGroup, String
+            stubsModule) {
+        URI stubJarUri = findGrabbedStubJars(skipLocalRepo, stubRepositoryRoot, stubsGroup, stubsModule)
+        File unzippedStubsDir = unpackStubJarToATemporaryFolder(stubJarUri)
+        return unzippedStubsDir
+    }
+
+    private File unpackStubJarToATemporaryFolder(URI stubJarUri) {
+        File tmpDirWhereStubsWillBeUnzipped = createTempDirectory(STUB_RUNNER_TEMP_DIR_PREFIX).toFile()
+        tmpDirWhereStubsWillBeUnzipped.deleteOnExit()
+        log.debug("Unpacking stub from JAR [URI: ${stubJarUri}]")
+        use(ZipCategory) {
+            new File(stubJarUri).unzipTo(tmpDirWhereStubsWillBeUnzipped)
+        }
+        return tmpDirWhereStubsWillBeUnzipped
+    }
+
+    private URI findGrabbedStubJars(boolean skipLocalRepo, String stubRepositoryRoot, String stubsGroup, String stubsModule) {
+        Map depToGrab = [group: stubsGroup, module: stubsModule, version: LATEST_MODULE, transitive: false]
+        setDependencyResolver(skipLocalRepo)
+        return dependencyResolver.resolveDependency(stubRepositoryRoot, depToGrab)
+    }
+
+    private setDependencyResolver(boolean skipLocalRepo) {
+        if (this.dependencyResolver == null) {
```

Comment:
It is misleading - `skipLocalRepo` flag in `downloadAndUnpackStubJar` is silently ignored when already set in a constructor. Is it only for tests or there is a use case when we want to predefine dependencyResolver?